packages/cache/src/internal/cacheHttpClient.ts

```diff
@@ -143,21 +143,22 @@ export async function reserveCache(
   key: string,
   paths: string[],
   options?: InternalCacheOptions
-): Promise<number> {
+): Promise<ITypedResponse<ReserveCacheResponse>> {
   const httpClient = createHttpClient()
   const version = getCacheVersion(paths, options?.compressionMethod)
 
   const reserveCacheRequest: ReserveCacheRequest = {
     key,
-    version
+    version,
+    cacheSize: options?.cacheSize
   }
   const response = await retryTypedResponse('reserveCache', async () =>
```

Comment:
you can try using [retryHttpClientResponse](https://github.com/actions/toolkit/blob/b4639928698a6bfe1c4bdae4b2bfdad1cb75016d/packages/cache/src/internal/requestUtils.ts#L123)

Reply:
Extended `ITypedResponse` to include `message` for error message. This will allow us to keep using the same method with advanced Response Type. This also gives us additional functionality of having error message as `message` in all the other responses for other API calls.